### DISCUSSION TITLE
Should we drop support for Python 2?

### DISCUSSION THREAD

**Discussion by erezsh** [lark-parser/lark#874] [General]
It seems that general Python 2 use is at around 6% : jetbrains.com/lp/python-developers-survey-2020

Should we just drop support for it in Lark?

It would make the code-base cleaner and easier to maintain, and we could use the new features.

Is anyone here still using Python 2?

**Comment by ThatXliner** [comment]
They shouldn't

**Comment by jnwatson** [comment]
I'm a maintainer of another package that still supports Python 2. I'm starting to run into issues where CI and testing packages are starting to remove Python 2 support.

I think Python 2 packages days are numbered. It will be increasingly difficulty to maintain support.

**Reply by ThatXliner** [reply]
Just curious, why do you need Python 2 support?

**Comment by phorward** [comment]
I would drop support for upcoming versions. If somebody needs Lark for a project in Python2, an older version can be used.
